hphp/hack/src/h2tp/unparser/unparser.ml

```diff
@@ -839,6 +839,7 @@ let rec u_program v = u_of_list_spc u_def v
     | Float f -> u_pstring f
     | String (p, s) ->
       StrList [Str "\""; u_pstring (p, Php_escaping.escape s); Str "\""]
+    | String2 [elem] -> StrList [u_expr elem; Str ".\"\""]
```

Comment:
I would say either leave it like it was in this case (`"$a"`), or explicitly convert it to a string `(string)$a`.